toolbox/timecomparer/main.go

```diff
@@ -0,0 +1,551 @@
+	}
+	return timeArray, commandsBreakDown
+}
+
+// Convert minutes and sesconds ("0m1.061s") to float object.
+func convertStringToTime(timeString string) []float64 {
+	timeSlice := []float64{}
+	if strings.Contains(timeString, "m") {
+		timeSplit := strings.Split(timeString, "m")
+		minString := timeSplit[0]
+		fmt.Println("get minutes")
+		fmt.Println(minString)
+		minInt, err := strconv.Atoi(minString)
+		if err != nil {
+			fmt.Println("error parsing float minute")
+			timeSlice = append(timeSlice, -1.0)
+		} else {
+			minFloat := float64(minInt)
+			timeSlice = append(timeSlice, minFloat)
+		}
+		if len(timeSplit) > 1 {
+			secString := timeSplit[len(timeSplit)-1]
+			secValueString := strings.Split(secString, "s")[0]
+			fmt.Println("get seconds")
+			fmt.Println(secValueString)
+			secFloat, err := strconv.ParseFloat(secValueString, 64)
+			if err != nil {
+				fmt.Println("error parsing float second")
+				timeSlice = append(timeSlice, -1.0)
+			} else {
+				timeSlice = append(timeSlice, secFloat)
+			}
+		}
+	}
+	return timeSlice
+}
+
+// Extract time from runtime string with the format of "real	0m1.061s"
+func extractTime(runTimeString string) runPair {
+	timeSplit := strings.Fields(runTimeString)
+	length := len(timeSplit)
+	var ind int
+	for ind = length - 1; ind > 0; ind-- {
+		if strings.Compare(timeSplit[ind], "") != 0 {
+			break
+		}
+	}
+	timePart := timeSplit[ind]
+	timeSlice := convertStringToTime(timePart)
+
+	for ind = 0; ind < length; ind++ {
+		if strings.Compare(timeSplit[ind], "") != 0 {
+			break
+		}
+	}
+	namePart := timeSplit[ind]
+
+	rPair := runPair{
+		name:       namePart,
+		time:       timeSlice,
+		stringTime: timePart,
+	}
+
+	return rPair
+
+}
+
+// Compute average user/real/sys time to compare with others.
+func getAverageTime(realTime, userTime, sysTime runPair) float64 {
+	averageSlice := []float64{}
+	realT := realTime.time
+	userT := userTime.time
+	sysT := sysTime.time
+
+	for i := 0; i < 2; i++ {
+		averagePart := (realT[i] + userT[i] + sysT[i]) / 3
+		averageSlice = append(averageSlice, averagePart)
+	}
+	average := averageSlice[0]*60 + averageSlice[1]
+
+	return average
+}
+
+// Get error contents of files to update command-time map with commands and path to build logs.
+func (tc *TimeComparer) findTimeCommands(
+	ctx context.Context, filePaths []string, commandToTime map[string]runCombination) map[string]runCombination {
+	timeCommand := regexp.MustCompile(`^time (.*);`)
+	realOutput := regexp.MustCompile(`^real(\s)*([0-9])*m([0-9])*\.([0-9])*s`)
+	userOutput := regexp.MustCompile(`^user(\s)*([0-9])*m([0-9])*\.([0-9])*s`)
+	sysOutput := regexp.MustCompile(`^sys(\s)*([0-9])*m([0-9])*\.([0-9])*s`)
+	for _, filePath := range filePaths {
+		fileSlice, err := tc.query(ctx, filePath)
+		if err != nil {
+			continue
+		}
+		for start := 0; start < len(fileSlice); start++ {
+			line := fileSlice[start]
+			if timeCommand.MatchString(line) {
+				for end := start; end < len(fileSlice)-2; end++ {
```

Comment:
What about the nested time commands
time do_a
time_do_b
....
real for_b
real for_a

What about using stack?